Model/Product/Variation/Builder.php

```diff
@@ -0,0 +1,216 @@
+        $this->localeDate = $localeDate;
+    }
+
+    /**
+     * @param Product $product
+     * @param NostoProduct $
+     * @param $nostoProduct
+     * @param Store $store
+     * @param Group $group
+     * @return Variation
+     */
+    public function build(
+        Product $product,
+        NostoProduct $nostoProduct,
+        Store $store,
+        Group $group
+    ) {
+        $variation = new Variation();
+        try {
+            $variation->setVariationId($group->getCode());
+            $variation->setAvailability($nostoProduct->getAvailability());
+            $variation->setPrice($this->getLowestVariationPrice($product, $group, $store));
+            $listPrice = $this->nostoCurrencyHelper->convertToTaggingPrice(
+                $this->nostoPriceHelper->getProductDisplayPrice(
+                    $product,
+                    $store
+                ),
+                $store
+            );
+            $variation->setListPrice($listPrice);
+            $variation->setPriceCurrencyCode($nostoProduct->getPriceCurrencyCode());
+        } catch (\Exception $e) {
+            $this->logger->exception($e);
+        }
+
+        $this->eventManager->dispatch(
+            'nosto_variation_load_after',
+            [
+                'variation' => $variation,
+                'magentoProduct' => $product
+            ]
+        );
+        return $variation;
+    }
+
+    /**
+     * @param Product $product
+     * @param Group $group
+     * @return float
+     */
+    private function getLowestVariationPrice(Product $product, Group $group, Store $store)
+    {
+        // If product is configurable, the parent has no customer group price. Get SKU with lowest price
+        if ($product->getTypeInstance() instanceof ConfigurableType) {
+            $product = $this->getMinPriceSku($product, $group, $store);
+        }
+        // Only returns the SKU price if it's lower than final price
+        // Merchant can have a fixed customer group price that is higher than the product
+        // price with a catalog price discount rule applied.
+        // This is normal Magento 2 behaviour
+        foreach ($product->getTierPrices() as $price) {
+            if ($price->getCustomerGroupId() === $group->getId()
+                && $price->getValue() < $product->getFinalPrice()
+            ) {
+                return $price->getValue();
+            }
+        }
+        // If no tier prices, there's no customer group pricing for this product
+        // or it's higher than final price with catalog price rule discount
+        return $product->getFinalPrice();
+    }
+
+    /**
+     * Returns the SKU|Product object with the lowest price.
+     *
+     * @param MageProduct $product
+     * @param Group $group
+     * @param Store $store
+     * @return MageProduct
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function getMinPriceSku(Product $product, Group $group, Store $store)
```

Comment:
This method should be used for the get list price as well so that the "discount" is accurate.  For example a configurable product has 2 skus. sku X has a list price 100, 20% off. Sku Y has a list price 90, no discount. When the reco shows the product, the discount should be 20%, not (90 - 80) / 90.

Reply:
Keep as discussed face to face.